core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java

```diff
@@ -111,11 +113,16 @@ public static EnvelopePart generateTimes(
             double[] positions,
             double[] speeds
     ) {
-        return new EnvelopePart(
-                new HashMap<>(),
+        var envelopeProfile = EnvelopeProfile.BRAKING;
+        if (areSpeedsEqual(speeds[0], speeds[speeds.length - 1])) {
+            envelopeProfile = EnvelopeProfile.CONSTANT_SPEED;
+        } else if (speeds[0] < speeds[speeds.length - 1]) {
+            envelopeProfile = EnvelopeProfile.ACCELERATING;
+        }
```

Comment:
This does not work reliably: what matters is whether the train is _trying_ to accelerate or hold speed, as it can be decelerating but not _braking_

Reply:
Agreed. But there are too many cases to work with. This method is only used in unit tests, and will always fail without a default EnvelopeProfile. I could move it into a static method in a TestHelper, would that be fine with you ?

Reply:
This specific overload is only used in unit tests, I suggest we move it to a test helper class and keep this (approximate) logic there

Reply:
> I could maybe move it into a static method in a TestHelper, would that be fine with you ?

That would do it, thanks